### PR TITLE
added go vet methods testing

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@liggitt This adds `go tools vet -methods`

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
re[test]

**Comment by stevekuznetsov** [comment]
re[test]

File: pkg/cmd/cli/cmd/edit.go [hunk 1]
```diff
@@ -286,7 +287,7 @@ func (h *editHeader) WriteTo(w io.Writer) error {
 		}
 		fmt.Fprintln(w, "#")
 	}
-	return nil
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/edit.go]: Hmm… the point of this is to comply with a common interface. The returned number should be the number of bytes written, I think

**Comment by stevekuznetsov** [comment]
@liggitt responded to comment

**Comment by liggitt** [comment]
The Fprint* functions already return length… just add up the returned value.

**Comment by stevekuznetsov** [comment]
@liggitt Honestly the other implementation was quite a lot cleaner ... thoughts? I can revert.

File: pkg/cmd/cli/cmd/edit.go [hunk 2]
```diff
@@ -268,7 +268,8 @@ type editHeader struct {
 }
 
 // WriteTo outputs the current header information into a stream
-func (h *editHeader) WriteTo(w io.Writer) error {
+func (h *editHeader) WriteTo(w io.Writer) (int64, error) {
+	var length int64
 	fmt.Fprint(w, `# Please edit the object below. Lines beginning with a '#' will be ignored,
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/edit.go]: capture the length written here

File: pkg/cmd/cli/cmd/edit.go [hunk 3]
```diff
@@ -276,16 +277,33 @@ func (h *editHeader) WriteTo(w io.Writer) error {
 `)
 	for _, r := range h.reasons {
 		if len(r.other) > 0 {
-			fmt.Fprintf(w, "# %s:\n", r.head)
+			lenBytes, err := fmt.Fprintf(w, "# %s:\n", r.head)
+			if err != nil {
+				return length, err
+			}
+			length = length + int64(lenBytes)
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/edit.go]: append the returned length regardless of the error condition

**Comment by stevekuznetsov** [comment]
@liggitt Responded to comments, but again I think using the buffer was quite a lot cleaner.

**Comment by liggitt** [comment]
I think you're right... I didn't realize this was the edit buffer, so the output will be small

**Comment by stevekuznetsov** [comment]
@liggitt Changed back to `byes.Buffer`

File: pkg/cmd/cli/cmd/edit.go [hunk 4]
```diff
@@ -268,24 +268,27 @@ type editHeader struct {
...
+
 		for _, o := range r.other {
-			fmt.Fprintf(w, "# * %s\n", o)
+			buffer.WriteString(fmt.Sprintf("# * %s\n", o))
 		}
-		fmt.Fprintln(w, "#")
+		buffer.WriteString("#")
 	}
-	return nil
+	fmt.Fprintln(w, buffer.String())
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/edit.go]: `return fmt.Fprintln(w, buffer.String())`

Comment (liggitt) [inline pkg/cmd/cli/cmd/edit.go]: nm, I see the int64 cast is needed

**Comment by liggitt** [comment]
LGTM, [merge]

**Comment by stevekuznetsov** [comment]
@liggitt whichever merges first will break the next one!

**Comment by liggitt** [comment]
I know... stop putting the first flag on the same line as the command :)

**Comment by stevekuznetsov** [comment]
@liggitt Even the flags on other lines were breaking! No way around it afaik

**Pull request merged**